packages/node-runtime-worker-thread/src/worker-runtime.ts

```diff
@@ -97,6 +97,7 @@ const workerRuntime: WorkerRuntime = {
 
     let interrupted = true;
     let evaluationPromise: void | Promise<RuntimeEvaluationResult>;
+    const previousRequireCache = Object.keys(require.cache);
```

Comment:
Because of webpack I'm pretty sure we'll have to do something similar to this to get to the real `require.cache`:

```suggestion
    const getRequireCache = () => (typeof __webpack_require__ === 'function' ? __non_webpack_require__ : require).cache;
    const previousRequireCache = Object.keys(getRequireCache());
```

In the meantime I'll open a ticket (and maybe will just do it when I have a minute) to replace webpack build here with something less aggressive with it's custom module system

Reply:
Hm … but wouldn’t we *want* the webpack equivalent of the require cache in this case, since it’s webpack’ed modules that we’re concerned about? (I did test this with the webpack’ed version as generated by `npm run pretest-ci`, fwiw)

Reply:
If I'm understanding the flow correctly here (and I definitely might be wrong!) I think we do want to get to the node require. Reason being that this runtime uses `ElectronRuntime` to evaluate the code, and we explicitly pass node require to the vm context there (doing the same check when creating an instance):

https://github.com/mongodb-js/mongosh/blob/d562a20bba65d598036db65e2a68ee2b644fd60f/packages/browser-runtime-electron/src/electron-runtime.ts#L37-L44

So when babel finished modifying the code and eventually passes it to the eval, it's the node require it will refer to in that context, right?

Reply:
> So when babel finished modifying the code and eventually passes it to the eval, it's the node require it will refer to in that context, right?

Yes, exactly – but the errors we’re seeing are because we might already be interrupting while babel is doing its work, which is what causes the problem here.

You’re right in that this would not fix code that runs in the REPL, and uses a `require()` call that gets interrupted, but that’s something that’s also ‘broken’ in the CLI.

Reply:
Ohhhhh, yes, okay, you're totally right here, now it clicked, evaluate is not _just_ evaluating! Sorry, took several attempts to wrap my head around it 😆

Reply:
Unresolving because seems like it's [still happening](https://mongodb.slack.com/archives/GJ2CBBY86/p1617869984107500) and looking closer at [what `babel explode` is](https://github.com/babel-utils/babel-explode-module) it seems like it's a thing that we import inside the runtime (so with the real require, not a webpack one)

I guess that means that we actually want to clean caches of both requires, what do you think?